### ISSUE TITLE
Enabling time traveling in documentation

### ISSUE THREAD

**Issue by stefanv** [scikit-image/scikit-image#3540]
See https://github.com/numpy/numpy/pull/12097/files

**Comment by grlee77** [comment]
Hi @stefanv, I moved this back from Discussions. is this something different from the version selection we have currently in the docs? (not sure if we should reopen)

**Comment by jni** [comment]
Yes, this should be reopened. The idea here is to travel to the *same page* in newer/older docs. The current links all take you to the docs index, and you have to navigate back to where you were.

The PyData sphinx theme does this automagically in the latest version, though I don't know how the config there works.
